---
layout: page
title: Publication
show-avatar: false
---

_First, correponding author papers marked in_ <span style="color:darkred">red</span>.<br>
_(^co-first author, * corresponding author)_

## 2019

 - <span style="color:darkred">Gao GF^, Parker JS^, Reynolds SM^, Silva TC^, Wang LB^, **Zhou W**^, et al. Before and After: A Comparison of Legacy and Harmonized TCGA Data at the Genomic Data Commons. **<u>Cell Systems</u>** (2019) *Co-first authors with equal contribution are ordered alphabetically. Zhou W leads the harmonization of DNA methylation data*.</span>

## 2018

 - Corces MR, Granja JM, Shams S, Louie BH, Seoane JA, **Zhou W**, Silva TC, Groeneveld C, Wong CK, Cho SW, Satpathy AT, Mumbach MR, Hoadley KA, Robertson AG, Sheffield NC, Felau I, Castro MAA, Berman BP, Staudt LM, Zenklusen JC, Laird PW, Curtis C, The Cancer Genome Atlas Research Network, Greenleaf WJ, Chang HY. The chromatin accessibility landscape of primary human cancers. **<u>Science</u>** (2018)
 - <span style="color:darkred">Carpenter BL^, **Zhou W**^, Madaj Z, DeWitt AK, Ross JP, Grønbæk K, Liang G, Clark SJ, Molloy P, Jones PA. Mother-child transmission of epigenetic information by variable imprinting. **<u>Proceedings of National Academy of Sciences</u>** (2018)</span>
 - Liu M, Thomas SL, DeWitt AK, **Zhou W**, Madaj ZB, Ohtani H, Baylin SB, Liang G, Jones PA. DNA and histone methyltransferase inhibitors cooperate to increase viral mimicry in cancer cells. **<u>Cancer Research</u>** (2018)
 - <span style="color:darkred">**Zhou W***, Triche Jr TJ., Laird PW and Shen H*. SeSAMe: Reducing artifactual detection of DNA methylation by Infinium BeadChips in genomic deletions. **<u>Nucleic Acids Research</u>** (2018)</span>
 - Ohtani H, Liu M, **Zhou W**, Liang G, Jones PA. Switching roles for DNA and histone methylation depend on evolutionary ages of human endogenous retroviruses. **<u>Genome Research</u>**, Jul 3. pii: gr.234229.118 (2018)
 - Shen H, Shih J, Hollern DP, Wang L, Bowlby R, Tickoo SK, Þórsson V, Mungall AJ, Newton Y, Hegde AM, Armenia J, Sánchez-Vega F, Pluta J, Pyle LC, Mehra R, Reuter VE, Godoy G, Jones J, Shelley CS, Feldman D, Vidal DO, Lessel D, Kulis T, Cárcano FM, Leraas KM, Lichtenberg TM, Brooks D, Cherniack AD, Cho J, Heiman DI, Kasaian K, Liu M, Noble MS, Xi L, Zhang H, **Zhou W**, ZenKlusen JC, Hutter CM, Felau I, Zhang J, Schultz N, Getz G, Meyerson M, Stuart JM, The Cancer Genome Atlas Research Network, Akbani R, Wheeler D, Laird PW, Nathanson KL, Cortessis VK, and Hoadley KA. Integrated molecular characterization of testicular germ cell tumors. **<u>Cell Reports</u>** 23 (11), 3392-3406 (2018)
 - Liu M, Zhang L, Li H, Hinoue T, **Zhou W**, Ohtani H, El-Khoueiry, A, Daniels J, O’Connell C, Dorff TB, Lu Q, Weisenberger DJ, Liang G. Integrative epigenetic analysis reveals therapeutic targets to the DNA methyltransferase inhibitor SGI-110 in hepatocellular carcinoma. **<u>Hepatology</u>**, May 18 (2018)
 - <span style="color:darkred">**Zhou W**^, Dinh HQ^, Ramjan Z, Weisenberger DJ, Nicolet CM, Shen H*, Laird PW*, Berman BP*. DNA methylation loss in late-replicating domains is linked to mitotic cell division. **<u>Nature Genetics</u>** 50:591–602, doi:10.1038/s41588-018-0073-4 (2018)</span>
 - Sanchez-Vega F, Mina M, Armenia J, Chatila WK, Luna A, La KC, Dimitriadoy S, Liu DL, Kantheti HS, Saghafinia S, Chakravarty D, Daian F, Gao Q, Bailey MH, Liang WW, Foltz SM, Shmulevich I, Ding L, Heins Z, Ochoa A, Gross B, Gao J, Zhang H, Kundra R, Kandoth C, Bahceci I, Dervishi L, Dogrusoz U, **Zhou W**, Shen H, Laird PW, Way GP, Greene CS, Liang H, Xiao Y, Wang C, Iavarone A, Berger AH, Bivona TG, Lazar AJ, Hammer GD, Giordano T, Kwong LN, McArthur G, Huang C, Tward AD, Frederick MJ, McCormick F, Meyerson M; Cancer Genome Atlas Research Network, Van Allen EM, Cherniack AD, Ciriello G, Sander C, Schultz N. The molecular landscape of oncogenic signaling pathways in The Cancer Genome Atlas. **<u>Cell</u>** 173(2):321-337.e10 (2018)
 - Thorsson V, Gibbs DL, Brown SD, Wolf D, Bortone DS, Ou Yang TH, Porta-Pardo E, Gao GF, Plaisier CL, Eddy JA, Ziv E, Culhane AC, Paull EO, Sivakumar IKA, Gentles AJ, Malhotra R, Farshidfar F, Colaprico A, Parker JS, Mose LE, Vo NS, Liu J, Liu Y, Rader J, Dhankani V, Reynolds SM, Bowlby R, Califano A, Cherniack AD, Anastassiou D, Bedognetti D, Rao A, Chen K, Krasnitz A, Hu H, Malta TM, Noushmehr H, Pedamallu CS, Bullman S, Ojesina AI, Lamb A, **Zhou W**, Shen H, Choueiri TK, Weinstein JN, Guinney J, Saltz J, Holt RA, Rabkin CE; Cancer Genome Atlas Research Network, Lazar AJ, Serody JS, Demicco EG, Disis ML, Vincent BG, Shmulevich L. The Immune Landscape of Cancer. **<u>Immunity</u>** S1074-7613(18)30121-3 (2018)

## 2017

 - Martin KR, **Zhou W**, Bowman MJ, Shih J, Au KS, Dittenhafer-Reed KE, Sisson KA, Koeman J, Weisenberger DJ, Cottingham SL, DeRoos ST, Devinsky O, Winn ME, Cherniack AD, Shen H, Northrup H, Krueger DA, MacKeigan JP. Genomic landscape of Tuberous Sclerosis Complex. **<u>Nature Communication</u>** 8:15816 (2017)
 - <span style="color:darkred">**Zhou W**, Laird PW, Shen H*. Comprehensive characterization, annotation and innovative use of Infinium DNA Methylation BeadChip probes. **<u>Nucleic Acids Research</u>** 45(4) e22-e22 (2017)</span>
 - Chong Z, Ruan J, Gao M, **Zhou W**, Chen T, Fan X, Ding L, Lee AY, Boutros P, Chen J, Chen K. novoBreak: glocal assembly for breakpoint detection in cancer genomes. **<u>Nature Methods</u>** 14(1) 65-67 (2017)
 - Guo Y, Vickers K, Xiong Y, Zhao S, Sheng Q, Zhang P, **Zhou W**, Flynn CR. Comprehensive evaluation of extracellular small RNA isolation methods from serum in high throughput sequencing. **<u>BMC Genomics</u>** 5(5), 485-497 (2017)

## 2016

 - <span style="color:darkred">Liu M^, Ohtani H^, **Zhou W**^, Ørskov AD^, Charlet J, Zhang YW, Shen H, Baylin SB, Liang G, Grønbæk K, Jones PA*. Vitamin C increases viral mimicry induced by 5-aza-2'-deoxycytidine. **<u>Proceedings of National Academy of Sciences</u>** 113(37):10238-10244 (2016) (F1000-recommended)</span>
 - Chen T, Wang Z, **Zhou W**, Chong Z, Meric-Bernstam F, Mills GB, Chen K. Hotspot mutations delineating diverse mutational signatures and biological utilities across cancer types. **<u>BMC Genomics</u>** 17 (Suppl 2):394 (2016)

## 2015

 - <span style="color:darkred">**Zhou W**^, Chen T^, Chong Z, Rohrdanz MA, Melott JM, Wakefield C, Zeng J, Weinstein JN, Meric-Bernstam F, Mills GB, Chen K*. TransVar: a multi-level variant annotator for precision genomics. **<u>Nature Methods</u>** 12:11 (2015)</span>
 - Sudmant PH, Rausch T, Gardner EJ, Handsaker RE, Abyzov A, Huddleston J, Zhang Y, Ye K, Jun G, Fritz MH, Konkel MK, Malhotra A, Stütz AM, Shi X, Casale FP, Chen J, Hormozdiari F, Dayama G, Chen K, Malig M, Chaisson MJP, Walter K, Meiers S, Kashin S, Garrison E, Auton A, Lam HYK, Mu XJ, Alkan C, Antaki D, Bae T, Cerveira E, Chines P, Chong Z, Clarke L, Dal E, Ding L, Emery S, Fan X, Gujral M, Kahveci F, Kidd JM, Kong Y, Lameijer EW, McCarthy S, Flicek P, Gibbs RA, Marth G, Mason CE, Menelaou A, Muzny DM, Nelson BJ, Noor A, Parrish NF, Pendleton M, Quitadamo A, Raeder B, Schadt EE, Romanovitch M, Schlattl A, Sebra R, Shabalin AA, Untergasser A, Walker JA, Wang M, Yu F, Zhang C, Zhang J, Zheng-Bradley X, **Zhou W**, Zichner T, Sebat J, Batzer MA, McCarroll SA; 1000 Genomes Project Consortium, Mills RE, Gerstein MB, Bashir A, Stegle O, Devine SE, Lee C, Eichler EE, Korbel JO. An integrated map of structural variation in 2,504 human genomes. **<u>Nature</u>** 526 (7571), 75-81 (2015)
 - <span style="color:darkred">**Zhou W**, Zhao H, Chong Z, Mark RJ, Eterovic AK, Meric-Bernstam F, Chen K. ClinSeK: a targeted variant characterization framework for clinical sequencing. **<u>Genome Medicine</u>** 7.1:1-9 (2015)</span>
 - Chen K, Meric-Bernstam F, Zhao H, Zhang Q, Ezzeddine N, Tang LY, Qi Y, Mao Y, Chen T, Chong Z, **Zhou W**, Zheng X, Johnson A, Aldape KD, Routbort MJ, Luthra R, Kopetz S, Davies MA, de Groot J, Moulder S, Vinod R, Farhangfar CJ, Shaw KM, Mendelsohn J, Mills GB, Eterovic AK. Clinical actionability enhanced through deep targeted sequencing of solid tumors. **<u>Clinical Chemistry</u>** 61(3):544-53 (2015)
 - Liu S, Meric-Bernstam F, Parinyanitikul N, Wang B, Eterovic AK, Zheng X, Gagea M, Chavez-MacGregor M, Ueno NT, Lei X, **Zhou W**, Nair L, Tripathy D, Brown PH, Hortobagyi GN, Chen K, Mendelsohn J, Mills GB, Gonzalez-Angulo AM. Functional consequence of the MET-T1010I polymorphism in breast cancer. **<u>Oncotarget</u>** 6(5):2604 (2015)

## 2014

 - <span style="color:darkred">**Zhou W***, Chen T, Zhao H, Eterovic AK, Meric-Bernstam F, Mills GB, Chen K*. Bias from removing read duplication in ultra-deep sequencing experiments. **<u>Bioinformatics</u>** 30.8:1073-1080 (2014)</span>
 - Fan X, **Zhou W**, Chong Z, Nakhleh L, Chen K. Towards accurate characterization of clonal heterogeneity based on structural variation. **<u>BMC Bioinformatics</u>** 15:299 (2014)

## 2013

 - Berestovsky N, **Zhou W**, Nagrath D, Nakhleh L. Modelling integrated cellular machinery using hybrid Petri-Boolean networks. **<u>PLoS Computational Biology</u>** 9.11:e1003306 (2013)

## 2012

 - <span style="color:darkred">**Zhou W*** and Nakhleh L*. Convergent evolution of modularity in metabolic networks through different community structures. **<u>BMC Evolutionary Biology</u>** 12.1:181 (2012)</span>
 - <span style="color:darkred">**Zhou W*** and Nakhleh L*. Quantifying and assessing the effect of chemical symmetry in metabolic pathways. **<u>Journal of Chemical Information and Modelling</u>** 52.10:2684-2696 (2012)</span>
    
## 2011

 - <span style="color:darkred">**Zhou W*** and Nakhleh L*. The strength of chemical linkage as a criterion for pruning metabolic graphs. **<u>Bioinformatics</u>** 14:1957-1963 (2011)</span>
 - <span style="color:darkred">**Zhou W*** and Nakhleh L*. Properties of metabolic graphs: biological organization or representation artifacts? **<u>BMC Bioinformatics</u>** 12:132 (2011)</span>

## Other publications

 - Wang Y, Xu X, Maglic D et al. (**Zhou W** included in the Cancer Genome Atlas Network) Comprehensive Molecular Characterization of the Hippo Signaling Pathway in Cancer. **<u>Cell Reports</u>** Oct 30;25(5):1304-1317.e5 (2018) 
 - Korkut A, Zaidi S, Kanchi RS et al. (**Zhou W** included in the Cancer Genome Atlas Network) A Pan-Cancer Analysis Reveals High-Frequency Genetic Alterations in Mediators of Signaling by the TGF-β Superfamily. **<u>Cell Systems</u>** Oct 24;7(4):422-437.e7 (2018)
 - Kahles A, Lehmann KV et al. (**Zhou W** included in the Cancer Genome Atlas Network) Comprehensive Analysis of Alternative Splicing Across Tumors from 8,705 Patients. **<u>Cancer Cell</u>** Aug 1. pii: S1535-6108(18)30306-4 (2018).
 - Hoadley KA, Yau C, Hinoue T, Wolf DM, Lazar AJ, Drill E, Shen R, Taylor AM, Cherniack AD, Thorsson V, Akbani R, Bowlby R, Wong CK et al., (**Zhou W** included in the Cancer Genome Atlas Network) Cell-of-Origin Patterns Dominate the Molecular Classification of 10,000 Tumors from 33 Types of Cancer. **<u>Cell</u>** 173(2):291-304.e6 (2018)
 - Malta TM, Sokolov A et al., (**Zhou W** included in the Cancer Genome Atlas Network) Machine Learning Identifies Stemness Features Associated with Oncogenic Dedifferentiation. **<u>Cell</u>** 173(2):338-354.e15 (2018)
 - Ding L, Bailey MH, Porta-Pardo E et al., (**Zhou W** included in the Cancer Genome Atlas Network) Perspective on Oncogenic Processes at the End of the Beginning of Cancer Genomics. **<u>Cell</u>** 173(2):305-320.e10 (2018)
 - Bailey MH, Tokheim C, Porta-Pardo E et al., (**Zhou W** included in the Cancer Genome Atlas Network) Comprehensive Characterization of Cancer Driver Genes and Mutations. **<u>Cell</u>** 173(2):371-385.e18 (2018)
 - Liu J et al., (**Zhou W** included in the Cancer Genome Atlas Network) An Integrated TCGA Pan-Cancer Clinical Data Resource to Drive High-Quality Survival Outcome Analytics. **<u>Cell</u>** 173(2):400-416.e11 (2018)
 - Chen H, Li C et al., (**Zhou W** included in the Cancer Genome Atlas Network) A Pan-Cancer Analysis of Enhancer Expression in Nearly 9000 Patient Samples. **<u>Cell</u>** 173(2):386-399.e12 (2018)
 - Huang KL et al., (**Zhou W** included in the Cancer Genome Atlas Network) Pathogenic Germline Variants in 10,389 Adult Cancers. **<u>Cell</u>** 173(2):355-370.e14 (2018)
 - Liu Y, Sethi NS, Hinoue T, Schneider BG et al., (**Zhou W** included in the Cancer Genome Atlas Network) Comparative Molecular Analysis of Gastrointestinal Adenocarcinomas. **<u>Cancer Cell</u>** pii: S1535-6108(18)30114-4 (2018)
 - Wang Z, Bo Yang et al., (**Zhou W** included in the Cancer Genome Atlas Network) lncRNA Epigenetic Landscape Analysis Identifies EPIC1 as an Oncogenic lncRNA that interacts with MYC and Promotes Cell-Cycle Progression in Cancer. **<u>Cancer Cell</u>** pii: S1535-6108(18)30110-7 (2018)
 - Berger AC, Korkut A, Kanchi RS et al., (**Zhou W** included in the Cancer Genome Atlas Network) A Comprehensive Pan-Cancer Molecular Study of Gynecologic and Breast Cancers. **<u>Cancer Cell</u>** pii: S1535-6108(18)30119-3 (2018)
 - Taylor AM et al., (**Zhou W** included in the Cancer Genome Atlas Network) Genomic and Functional Approaches to Understanding Cancer Aneuploidy. **<u>Cancer Cell</u>** pii: S1535-6108(18)30111-9 (2018)
 - Ricketts CJ et al., (**Zhou W** included in the Cancer Genome Atlas Network) The Cancer Genome Atlas Comprehensive Molecular Characterization of Renal Cell Carcinoma. **<u>Cell Reports</u>** 23(1):313-326.e5 (2018)
 - Chiu HS, Somvashi S et al., (**Zhou W** included in the Cancer Genome Atlas Network) Pan-Cancer Analysis of lncRNA Regulation Supports Their Targeting of Cancer Genes in Each Tumor Context. **<u>Cell Reports</u>** 23(1):297-312.e12 (2018)
 - Seiler M, Peng S et al., (**Zhou W** included in the Cancer Genome Atlas Network) Somatic Mutational Landscape of Splicing Factor Genes and Their Functional Consequences across 33 Cancer Types. **<u>Cell Reports</u>** 23(1):282-296.e4 (2018)
 - Jayasinghe RG, Song Cao et al., (**Zhou W** included in the Cancer Genome Atlas Network) Systematic Analysis of Splice-Site-Creating Mutations in Cancer. **<u>Cell Reports</u>** 23(1):270-281.e3 (2018)
 - Peng X, Chen Z, Farshidfar F et al., (**Zhou W** included in the Cancer Genome Atlas Network) Molecular Characterization and Clinical Relevance of Metabolic Expression Subtypes in Human Cancers. **<u>Cell Reports</u>** 23(1):255-269.e4 (2018)
 - Knijnenburg TA, Wang L, Zimmerman MT, Chambwe N et al., (**Zhou W** included in the Cancer Genome Atlas Network) Genomic and Molecular Landscape of DNA Damage Repair Deficiency across The Cancer Genome Atlas. **<u>Cell Reports</u>** 23(1):239-254.e6 (2018)
 - Gao Q, Liang WW, Foltz SM et al., (**Zhou W** included in the Cancer Genome Atlas Network) Driver Fusions and Their Implications in the Development and Treatment of Human Cancers. **<u>Cell Reports</u>** 23(1):227-238.e3 (2018)
 - Ge Z, Leighton JS et al., (**Zhou W** included in the Cancer Genome Atlas Network) Integrated Genomic Analysis of the Ubiquitin Pathway across Cancer Types. **<u>Cell Reports</u>** 23(1):213-226.e3 (2018)
 - Campbell JD, Yao C, Bowlby R, Liu Y, Brennan K, Fan H, Taylor AM, Wang C, Walter V, Akbani R, Byer LA, Crighton CJ, Coarfa C et al., (**Zhou W** included in the Cancer Genome Atlas Network) Genomic, Pathway Network, and Immunologic Features Distinguishing Squamous Carcinomas. **<u>Cell Reports</u>** 23(1):194-212.e6 (2018)
 - Saltz J et al., (**Zhou W** included in the Cancer Genome Atlas Network) Spatial Organization and Molecular Correlation of Tumor-Infiltrating Lymphocytes Using Deep Learning on Pathology Images. **<u>Cell Reports</u>** 23(1):181-193.e7 (2018)
 - Way GP et al., (**Zhou W** included in the Cancer Genome Atlas Network) Machine Learning Detects Pan-cancer Ras Pathway Activation in The Cancer Genome Atlas. **<u>Cell Reports</u>** 23(1):172-180.e3 (2018)
 - Schaub FX et al., (**Zhou W** included in the Cancer Genome Atlas Network) Pan-cancer Alterations of the MYC Oncogene and Its Proximal Network across the Cancer Genome Atlas. **<u>Cell Systems</u>** 6(3):282-300.e2 (2018)
 - Ellrott K et al., (**Zhou W** included in the Cancer Genome Atlas Network) Scalable Open Science Approach for Mutation Calling of Tumor Exomes Using Multiple Genomic Pipelines. **<u>Cell Systems</u>** 6(3):271-281.e7 (2018)
 - Gönen M, Weir BA, Cowley GS, Vazquez F, Guan Y, Jaiswal A, Karasuyama M, Uzunangelov V, Wang T et al., (**Zhou W** included in Broad-DREAM Challenge) A community challenge for inferring genetic predictors of gene essentialities through analysis of a functional screen of cancer cell lines. **<u>Cell Systems</u>** S2405-4712(17)30392-7 (2017)
